src/sunsynk/sensors.py

```diff
@@ -38,11 +39,14 @@ def id(self) -> str:
     def reg_to_value(self, regs: RegType) -> ValType:
         """Return the value from the registers."""
         regs = self.masked(regs)
-        val: NumType = regs[0]
-        if len(regs) > 1:
-            val += regs[1] << 16
-        if self.factor < 0:  # Indicates this register is signed
-            val = signed(val, bits=16 * len(regs))
+        if len(regs) == 1:
+            val = regs[0]
+            if self.factor < 0:  # Indicates this register is signed
+                val = signed(val, bits=16)
+        elif len(regs) == 2:
+            val = make_32bit(regs[0], regs[1], signed=self.factor < 0)
```

Comment:
Can we combine signed & make_32bit?

```python
def unpack(*regs: int, *, signed: bool = True, bits: int=16) -> int:
```

Reply:
This iwll break on of the signed tests of today, but in those test, 32-bit must be split into 2x regs